js/components/App.js

```diff
@@ -148,6 +149,20 @@ class App extends React.Component {
               ref={component => this._gotRef(id, component)}
             />
           );
+        case WINDOWS.MILKDROP:
+          return (
+            <MilkdropWindow
+              ref={component => this._gotRef(id, component)}
+              title={w.title}
+              options={this.props.butterchurnOptions}
+              windowId={id}
+              onFocusedKeyDown={listener => this._emitter.on(id, listener)}
+              analyser={media.getAnalyser()}
+              isEnabledVisualizer={this.props.visualizerStyle === id}
+              playing={this.props.status === MEDIA_STATUS.PLAYING}
+              close={() => this.props.closeWindow(id)}
```

Comment:
I think everything here except ref and analyser could move into the milkdrop window. They could either be hard coded (title/windowId) or they could be derived from the redux state via connect. If you want to take a stab at that, cool. If not, I can followup.

Reply:
ah that makes sense, I attempted to move the redux stuff to the milkdrop window